/*
 * Copyright (c) Huawei Technologies Co., Ltd. 2025. All rights reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

syntax = "proto3";

package agent_plugin;
option go_package = "functionsystem/pkg/common/faas_common/grpc/pb/agent_plugin;agent_plugin";

import "google/protobuf/any.proto";

service PluginService {
  rpc Call(PluginRequest) returns (PluginResponse);
}

message PluginRequest {
  string traceID = 1;
  string requestID = 2;
  string method = 3;
  google.protobuf.Any payload = 4;
  map<string, string> metadata = 5;
}

message PluginResponse {
  string traceID = 1;
  string requestID = 2;
  int32 code = 3;
  string message = 4;
  google.protobuf.Any payload = 5;
}

// virtual env
message PrepareRequest {
  string language = 1;
  map<string, string> createOptions = 2;
}

message PrepareResponse {
  int32 code = 1;
  string message = 2;
  map<string, string> cmds= 3;
}

message ClearRequest {
  map<string, string> clearOptions = 1;
}

message ClearResponse {
  int32 code = 1;
  string message = 2;
}
